My lab develops fast Non-negative Matrix Factorization algorithms for applications in bioinformatics and beyond. Special interests include single-cell experiment dimensional reduction and integration, hybrid NNLS/graph-based learning, genome/phenome meta-association, and novel approaches to model regularization.

Ongoing C++ library development:
* Sparse Run-length Encoding for compressed storage of discrete sparse matrices
* Out-of-core dimension reduction with Non-negative Matrix Factorization
* New Non-negative Matrix Factorization methods and weighting algorithms

Ongoing R package development:
* A new "Seurat" object class occupying 10x less space in memory, and able to hold indefinitely many samples
* Fast and scalable spatially-aware dimensional reduction
* Graph-regularized Non-negative Matrix Factorization
* Integrative transfer learning with linked Non-negative Least Squares projections
